amip_interp/amip_interp.F90

```diff
@@ -265,31 +275,39 @@ module amip_interp_mod
 !!
 !> @ingroup amip_interp_mod
 interface amip_interp_new
-   module procedure amip_interp_new_1d
-   module procedure amip_interp_new_2d
+   module procedure amip_interp_new_1d_r4, amip_interp_new_1d_r8
+   module procedure amip_interp_new_2d_r4, amip_interp_new_2d_r8
 end interface
 
+!> Initialize amip_interp. To explicitly initialize the module in r4 mode or r8 mode,
+!! call either amip_interp_init_r4 or amip_interp_init_r8. amip_interp_init is an
+!! alias for amip_interp_init_r8.
+interface amip_interp_init
+  module procedure amip_interp_init_r8
+end interface
 
 !----- public data type ------
 
-!> @brief Contains information needed by the interpolation module (exchange_mod) and buffers data.
+!> @brief Contains information needed by the interpolation module (exchange_mod) and buffers
+!! data (r4_kind flavor).
 !> @ingroup amip_interp_mod
 type amip_interp_type
    private
-   type (horiz_interp_type) :: Hintrp, Hintrp2 ! add by JHC
-   real, allocatable        ::    data1(:,:),  data2(:,:)
-   type (date_type)         ::    Date1,       Date2
-   logical                  :: use_climo, use_annual
-   logical                  :: I_am_initialized=.false.
-end type
+   type (horiz_interp_type)              :: Hintrp, Hintrp2 ! add by JHC
+   class(*), dimension(:,:), allocatable :: data1, data2
```

Comment:
I see pointers are being used in the .inc file cause there are class(*)? ...
Since data1 and data2 are private, can it be 
``real(r4_kind), dimension(:,:) :: data1_r4, data2_r4``
      ``real(r8_kind), dimension(:,:) :: data1_r8, data2_r8``

Then in the .inc file, the these arrays can be accessed by Interp%DATA1_KIND_ where DAT1_KIND_ is defined as data1_r4 in the .fh file.  @rem1776  came up with the idea and we've been doing something like that for other modules.